packages/cmf/src/componentState.js

```diff
@@ -17,7 +17,12 @@ export function initState(props) {
 export function applyCallback(callback, name, id) {
 	return (dispatch, getState) => {
 		const newState = callback(getStateProps(getState(), name, id));
-		dispatch(actions.componentsActions.mergeComponentState(name, id, newState));
+		const componentState = actions.componentsActions.mergeComponentState(name, id, newState);
+		dispatch({
+			id,
+			type: `${name}.setState`,
+			cmf: { componentState },
+		});
```

Comment:
Refactor : can you try to merge this code with the part where it's used ? 
Reason : this is the same piece of code, just the state is the result of the callback.